infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -368,14 +361,18 @@ private SequencerStatus getPrimarySequencerStatus(Layout layout, PollReport poll
      */
     private void handleFailures(PollReport pollReport) {
         try {
-            Set<String> failedNodes = new HashSet<>(getNewFailures(pollReport));
-
             // These conditions are mutually exclusive. If there is a failure to be
             // handled, we don't need to explicitly fix the unfilled layout slot. Else we do.
             Layout layout = serverContext.copyManagementLayout();
+            Set<String> failedNodes = new TreeSet<>(recommendationEngine
+                    .failedServers(clusterStateContext.getClusterState(), layout));
+
+            log.info("[{}] StartFailure {}", serverContext.getLocalEndpoint(), failedNodes);
+            clusterStateContext.printClusterView();
+
             if (!failedNodes.isEmpty() || isCurrentLayoutSlotUnFilled(pollReport)) {
 
-                log.info("Detected changes in node responsiveness: Failed:{}, pollReport:{}",
+                log.info("Detected failed nodes in node responsiveness: Failed:{}, pollReport:{}",
```

Comment:
There is a possibility that we get to this because of unfilled slot not failed nodes.

Reply:
Yes and that is the expected behaviour. We then commit the same layout again with the new epoch.